src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,409 @@
+    // of Dawn and the other is validation errors from Dawn.
+    if (exceptionName && exceptionName.length !== 0) {
+      this.shouldThrow(exceptionName, () => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      });
+    } else {
+      this.expectValidationError(() => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      }, !success);
+    }
+  }
+}
+
+export const g = makeTestGroup(CopyImageBitmapToTextureTest);
+
+g.test('copyImageBitmapToTexture_source_imageBitmap_copy_view')
+  .params(poptions('closed', [true, false]))
+  .fn(async t => {
+    const { closed } = t.params;
+    const src = await t.getImageBitmapCopyView(
+      DEFAULT_WIDTH,
+      DEFAULT_HEIGHT,
+      { x: 0, y: 0 },
+      closed
+    );
+    const dst = t.getDefaultTextureCopyView('2d');
+    const copySize = { width: 1, height: 1, depth: 1 };
+    const exceptionName = closed ? 'InvalidStateError' : '';
+
+    t.testRun(src, dst, copySize, !closed, exceptionName);
+  });
+
+g.test('copyImageBitmapToTexture_destination_texture_copy_view')
+  .params(
+    params()
+      .combine(poptions('destroyed', [true, false]))
+      .combine(poptions('dimension', ['2d'] as const))
+      .combine(poptions('usage', kTextureUsages))
+      .combine(poptions('sampleCount', [1, 4]))
+      .combine(poptions('mipLevel', [0, 2]))
+      .combine(poptions('format', kAllTextureFormats))
+  )
+  .fn(async t => {
+    const { destroyed, dimension, usage, sampleCount, mipLevel, format } = t.params;
+    const src = await t.getDefaultImageBitmapCopyView();
+    const dst = t.getTextureCopyViewWithDefaultSize(
+      sampleCount,
+      dimension,
+      format,
+      usage,
+      mipLevel,
+      { x: 0, y: 0, z: 0 },
+      destroyed
+    );
+    const copySize = { width: 1, height: 1, depth: 1 };
+
+    let success = true;
+    let exceptionName = '';
+
+    if (destroyed) {
+      success = false;
+    }
+
+    if (!(usage & GPUTextureUsage.COPY_DST)) {
+      // Usage must have GPUTextureUsage.COPY_DST
+      success = false;
+    }
+
+    if (sampleCount > 1) {
+      // Dst texture cannot be multisampled due to imageBitmap.
+      success = false;
+    }
+
+    if (mipLevel > 1) {
+      // MipLevel value cannot exceed dst texture max mipLevel, which is 1 in tests.
+      success = false;
+    }
+
+    if (!kValidTextureFormatsForCopyIB2T.includes(format)) {
+      // Dst texture format needs to be supported in copyImageBitmapToTexture.
+      success = false;
+      exceptionName = 'TypeError';
+    }
+
+    t.testRun(src, dst, copySize, success, exceptionName);
+  });
+
+g.test('copyImageBitmapToTexture_copy_size')
+  .params(
+    params()
+      .combine(poptions('dimension', ['2d'] as const))
+      .expand(copySizeTestForDstTextureSize)
```

Comment:
We should also test at a miplevel, to check that the miplevel size is correctly taken into account.

Reply:
Done.